pyincore/analyses/pipelinerepaircost/pipelinerepaircost.py

```diff
@@ -0,0 +1,220 @@
+
+        # join damage, replacement cost, with original inventory
+        pipeline_df = pipeline_df.merge(pipeline_dmg_df, on="guid")
+        pipeline_df = pipeline_df.merge(replacement_cost, on="guid")
+        pipeline_set = pipeline_df.to_dict(orient="records")
+
+        user_defined_cpu = 1
+        if not self.get_parameter("num_cpu") is None and self.get_parameter("num_cpu") > 0:
+            user_defined_cpu = self.get_parameter("num_cpu")
+
+        num_workers = AnalysisUtil.determine_parallelism_locally(self, len(pipeline_set), user_defined_cpu)
+
+        avg_bulk_input_size = int(len(pipeline_set) / num_workers)
+        inventory_args = []
+        count = 0
+        inventory_list = list(pipeline_set)
+        while count < len(inventory_list):
+            inventory_args.append(inventory_list[count:count + avg_bulk_input_size])
+            count += avg_bulk_input_size
+
+        repair_costs = self.pipeline_repair_cost_concurrent_future(self.pipeline_repair_cost_bulk_input, num_workers,
+                                                                   inventory_args)
+        self.set_result_csv_data("result", repair_costs, name=self.get_parameter("result_name") + "_repair_cost")
+
+        return True
+
+    def pipeline_repair_cost_concurrent_future(self, function_name, num_workers, *args):
+        """Utilizes concurrent.future module.
+
+        Args:
+            function_name (function): The function to be parallelized.
+            num_workers (int): Maximum number workers in parallelization.
+            *args: All the arguments in order to pass into parameter function_name.
+
+        Returns:
+            list: A list of ordered dictionaries with pipeline repair cost values and other data/metadata.
+
+        """
+
+        output = []
+        with concurrent.futures.ProcessPoolExecutor(max_workers=num_workers) as executor:
+            for ret1 in executor.map(function_name, *args):
+                output.extend(ret1)
+
+        return output
+
+    def pipeline_repair_cost_bulk_input(self, pipelines):
+        """Run analysis for multiple pipelines.
+
+        Args:
+            pipelines (list): Multiple pipelines from input inventory set.
+
+        Returns:
+            list: A list of ordered dictionaries with pipeline repair cost values and other data/metadata.
+
+        """
+        # read in the damage ratio tables
+        pipeline_dmg_ratios_csv = self.get_input_dataset("pipeline_dmg_ratios").get_csv_reader()
+        dmg_ratio_tbl = AnalysisUtil.get_csv_table_rows(pipeline_dmg_ratios_csv, ignore_first_row=False)
+
+        segment_length = self.get_parameter("segment_length")
+        if segment_length is None:
+            segment_length = 20  # 20 feet
+
+        diameter = self.get_parameter("diameter")
+        if diameter is None:
+            diameter = 20  # 20 inch
+
+        repair_costs = []
+
+        for pipeline in pipelines:
+            pipe_length = pipeline["length"]  # kilometre
```

Comment:
kilometer?

Reply:
hahaha maybe I am British!
jkjk will correct the typo